couchbase/admin/query/v1/query.proto

```diff
@@ -0,0 +1,111 @@
+syntax = "proto3";
+
+option go_package = "github.com/couchbase/goprotostellar/genproto/admin_query_v1;admin_query_v1";
+option ruby_package = "Couchbase::Protostellar::Generated::Admin::Query::V1";
+option php_namespace = "Couchbase\\Protostellar\\Generated\\Admin\\Query\\V1";
+option java_package = "com.couchbase.client.protostellar.admin.query.v1";
+option java_multiple_files = true;
+
+package couchbase.admin.query.v1;
+
+service QueryAdminService {
+  rpc GetAllIndexes(GetAllIndexesRequest) returns (GetAllIndexesResponse) {}
+
+  rpc CreatePrimaryIndex(CreatePrimaryIndexRequest)
+      returns (CreatePrimaryIndexResponse) {}
+  rpc CreateIndex(CreateIndexRequest) returns (CreateIndexResponse) {}
+
+  rpc DropPrimaryIndex(DropPrimaryIndexRequest)
+      returns (DropPrimaryIndexResponse) {}
+  rpc DropIndex(DropIndexRequest) returns (DropIndexResponse) {}
+
+  rpc BuildDeferredIndexes(BuildDeferredIndexesRequest)
+      returns (BuildDeferredIndexesResponse) {}
+}
+
+enum IndexType {
+  INDEX_TYPE_VIEW = 0;
+  INDEX_TYPE_GSI = 1;
+}
+
+enum IndexState {
+  INDEX_STATE_DEFERRED = 0;
+  INDEX_STATE_BUILDING = 1;
+  INDEX_STATE_PENDING = 2;
+  INDEX_STATE_ONLINE = 3;
+  INDEX_STATE_OFFLINE = 4;
+  INDEX_STATE_ABRIDGED = 5;
+  INDEX_STATE_SCHEDULED = 6;
+}
+
+message GetAllIndexesRequest {
+  optional string bucket_name = 1;
+  optional string scope_name = 2;
+  optional string collection_name = 3;
+}
+
+message GetAllIndexesResponse {
+  message Index {
+    string bucket_name = 1;
+    string scope_name = 2;
+    string collection_name = 3;
+    string name = 4;
+    bool is_primary = 5;
+    IndexType type = 6;
+    IndexState state = 7;
+    repeated string fields = 8;
+    optional string condition = 9;
+    optional string partition = 10;
+  }
```

Comment:
"keyspace" is missing.  This seems a tricky case.  I get the desire to keep the interface clean.  But it's also part of the public SDK API...

Reply:
Or do we just use bucket_name?

Reply:
Keyspace is indeed tricky, it's actually 'broken' for SDKs today.  It exists from the pre-collections era, where keyspace was always the bucket name.  We then added an explicit 'bucket_name' field which was just a mirror of the 'keyspace' field.  Now that we have collections though, keyspace is sometimes the bucket name, and sometimes the collection name, and has no real direct use, since we already 'break it down' to the individual fields as part of the SDK behaviour.  For Protostellar, we've opted to exclude that field, and SDKs should probably reconstruct it based on the collection_name/bucket_name fields (based on whether the collection_name field is populated).

Reply:
Right, I think that's reasonable.
Though you mention elsewhere that the collection_name field will always be default, even if it's just "_default".  So effectively we'll always bucket_name.  But that's probably want we want anyway

Reply:
Filed an issue to look at this asynchronously: https://couchbasecloud.atlassian.net/browse/ING-517